Automattic/jetpack
# Add: Sync  Plugin Updated event.

This PR adds test for plugin updates and plugin update failures.

#### Changes proposed in this Pull Request:

- Fix API for plugin updates so that it returns an error as expected. 
- Adds 
  - syncing of plugin updates on success fully plugin update and failure.
  - updates the ....wait for it....
 

#### Testing instructions:
* Do the tests pass? 
* make it so that plugin updates fail ( use a plugin like something below) 
```
<?php
/*
 * Plugin Name: Break Plugin Updates
*/

add_filter( 'upgrader_package_options', 'eb_break_plugin_updated' );
function eb_break_plugin_updated( $options ) {
        $options['package'] = 'http://example.com';
        return $options;
}
```
- try updating a plugin via the api (in calyspo) Do you notice an error? 
- try updating the plugin via the wp-admin interface, plugin.php and core-updates.php UI.
- try updating the plugin via the wp cli interface. `wp plugin update jetpack`
Do you get the expected failure event on .com? 

- Try updating the plugin do you get the event as expected on .com that tells you that a plugin updated? 

- Do you still get a plugin install event as expected?
